src/hash/hash_common_crypto.h

```diff
@@ -27,10 +29,21 @@ GIT_INLINE(int) git_hash_init(git_hash_ctx *ctx)
 	return 0;
 }
 
-GIT_INLINE(int) git_hash_update(git_hash_ctx *ctx, const void *data, size_t len)
+GIT_INLINE(int) git_hash_update(git_hash_ctx *ctx, const void *_data, size_t len)
 {
+	const unsigned char *data = _data;
```

Comment:
By the way, why did you add all those casts to the functions? Just curious

Reply:
xcode was whinging about a signed/unsigned mismatch on `CC_SHA1_Update`  so I decided to cast them to their proper types as defined by the implementations.